transport/tchannel/header.go

```diff
@@ -93,35 +93,27 @@ func readHeaders(format tchannel.Format, getReader func() (tchannel.ArgReader, e
 	return headers, r.Close()
 }
 
-func writeRequestHeaders(
-	ctx context.Context,
-	format tchannel.Format,
-	appHeaders map[string]string,
-	getWriter func() (tchannel.ArgWriter, error),
-) error {
-	headers := transport.NewHeadersWithCapacity(len(appHeaders))
-	// TODO: zero-alloc version
-
-	for k, v := range appHeaders {
-		headers = headers.With(k, v)
-	}
-
-	return writeHeaders(format, headers, getWriter)
-}
-
 // writeHeaders writes the given headers using the given function to get the
 // arg writer.
 //
 // This may be used with the Arg2Writer functions on OutboundCall and
 // InboundCallResponse.
 //
 // If the format is JSON, the headers are JSON encoded.
-func writeHeaders(format tchannel.Format, headers transport.Headers, getWriter func() (tchannel.ArgWriter, error)) error {
+func writeHeaders(format tchannel.Format, headers map[string]string, tracingBaggage map[string]string, getWriter func() (tchannel.ArgWriter, error)) error {
+	merged := make(map[string]string, len(headers)+len(tracingBaggage))
+	// tracingBaggage should not override application headers. So this goes first
```

Comment:
tracing baggage *should* override application headers i think, so it should go last